### PR TITLE
// implementation of soca to cice 

### LINKED ISSUE #989: The SOCA to CICE variable change needs to run in //
## Description
The "de-aggregation" of the 2D sea-ice variable into 3D model variable is:
1. Slow. The parallelization work was started but never finished. 
2. We can't do `srun -n 1` anymore on `Hera`, no idea why but a good reason to end my procrastination stint.

## Solution
Finish the parallelization work.

### PR DESCRIPTION
**Pull request by guillaumevernieres**
## Description
It's a 5pm Friday PR, don't get too exited.

### Issue(s) addressed
- fixes #989 

## Testing

How were these changes tested? **in containers at low-res**
What compilers / HPCs was it tested with? **gnu** **Not fully tested on Hera yet (intel)**
Are the changes covered by ctests? **Yes, running the ctest with 2 pes now**

### CODE REVIEW

**Comment by guillaumevernieres** [comment]
So far so good, ~20sec with 24 pe's/1 node. 
A note to self for Monday: rebuild the ufs to cycle and make sure the new cice restarts are OK.

File: src/soca/VariableChange/Soca2Cice/soca_ciceutils_mod.F90 [hunk 1]
```diff
@@ -73,59 +99,132 @@ subroutine soca_ciceutils_init(self, rst_filename, rst_out_filename)
...
+  call geom%f_comm%barrier()
+
+  ! gather 2D arrays along categories, snow and ice levels
+  do n = 1, self%ncat
+     ! aicen
+     call mpp_gather(self%isc, self%iec, self%jsc, self%jec, pelist, &
+          self%aicen(self%isc:self%iec,self%jsc:self%jec,n), &
+          var2d, is_root_pe)
+     if ( pe == mpp_root_pe()) glb%aicen(1:self%ni,1:self%nj,n) = var2d*glb%iceumask
+     call geom%f_comm%barrier()
```
Comment (travissluka) [inline src/soca/VariableChange/Soca2Cice/soca_ciceutils_mod.F90]: is there a reason there are `barrier()` calls here? Shouldn't be necessary

Comment (guillaumevernieres) [reply src/soca/VariableChange/Soca2Cice/soca_ciceutils_mod.F90]: Left over from debugging, I'll remove them.

**Review by travissluka — APPROVED** [review]

**Comment by guillaumevernieres** [comment]
Any chance this could be merged @travissluka ?

**Comment by travissluka** [comment]
> Any chance this could be merged @travissluka ?

no. I need 2 reviews. I lost admin privileges and can't override that

**Review by Dooruk — APPROVED** [review]
I am testing it today with GEOS CICE6 but I will preapprove it so it can be merged.

**Pull request merged**